src/Raven.Server/Rachis/RachisConsensus.cs

```diff
@@ -1054,6 +1054,9 @@ public BlittableJsonReaderObject SetTopology(ClusterOperationContext context, Cl
                 topologyJson.CopyTo(ptr);
             }
 
+            if (engine.ServerStore._lastClusterTopologyIndex < clusterTopology.Etag)
+                engine.ServerStore._lastClusterTopologyIndex = clusterTopology.Etag;
+
```

Comment:
I think we should move to the `OnDispose` of the tx
and check if the tx committed.

Reply:
done